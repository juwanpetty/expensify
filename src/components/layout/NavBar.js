import React from "react";
import styled from "styled-components";
import { NavLink } from "react-router-dom";

const NavBar = () => {
  return (
    <NavContainer>
      <Logo>Expensify</Logo>
      <Link to="/" exact activeStyle={ActiveLink}>
        Dashboard
      </Link>
      <Link to="/create" activeStyle={ActiveLink}>
        \Create Expense
      </Link>
      <Link to="/edit" activeStyle={ActiveLink}>
        Edit Expense
      </Link>
      <Link to="/help" activeStyle={ActiveLink}>
        Help
      </Link>
    </NavContainer>
  );
};

const Logo = styled.h1`
  margin: 20px 40px;
`;

const NavContainer = styled.nav`
  display: flex;
  align-items: center;
`;

const Link = styled(NavLink)`
  text-decoration: none;
  font-size: 18px;
  color: #9799a2;

  &:not(:last-child) {
    margin-right: 32px;
  }
`;

const ActiveLink = {
  color: "#303446"
};

export default NavBar;
