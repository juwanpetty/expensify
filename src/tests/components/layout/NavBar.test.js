import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import NavBar from "../../../components/layout/NavBar";

it("should render NavBar correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<NavBar />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
