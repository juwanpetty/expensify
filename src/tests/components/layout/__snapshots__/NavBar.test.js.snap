// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar correctly 1`] = `
<ForwardRef(styled.nav)>
  <ForwardRef(styled.h1)>
    Expensify
  </ForwardRef(styled.h1)>
  <ForwardRef(Styled(NavLink))
    activeStyle={
      Object {
        "color": "#303446",
      }
    }
    exact={true}
    to="/"
  >
    Dashboard
  </ForwardRef(Styled(NavLink))>
  <ForwardRef(Styled(NavLink))
    activeStyle={
      Object {
        "color": "#303446",
      }
    }
    to="/create"
  >
    Create Expense
  </ForwardRef(Styled(NavLink))>
  <ForwardRef(Styled(NavLink))
    activeStyle={
      Object {
        "color": "#303446",
      }
    }
    to="/help"
  >
    Help
  </ForwardRef(Styled(NavLink))>
</ForwardRef(styled.nav)>
`;
